### PR TITLE
Malloc bucket aware Array growth

### PR DESCRIPTION
**Pull request by Catfish-Man**

### CODE REVIEW

**Comment by Catfish-Man** [comment]
@swift-ci please smoke benchmark

**Comment by Catfish-Man** [comment]
@swift-ci please smoke test

**Comment by Catfish-Man** [comment]
@swift-ci please smoke test

**Comment by Catfish-Man** [comment]
@swift-ci please smoke benchmark

File: stdlib/public/core/ArrayShared.swift [hunk 1]
```diff
@@ -140,11 +140,14 @@ internal func _growArrayCapacity(_ capacity: Int) -> Int {
 
 @_alwaysEmitIntoClient
 internal func _growArrayCapacity(
-  oldCapacity: Int, minimumCapacity: Int, growForAppend: Bool
+  oldCapacity: Int, minimumCapacity: Int, elementSize: Int, growForAppend: Bool
 ) -> Int {
   if growForAppend {
     if oldCapacity < minimumCapacity {
-      // When appending to an array, grow exponentially.
+      if minimumCapacity * elementSize <= 128 {
```
Comment (stephentyrone) [inline stdlib/public/core/ArrayShared.swift]: What's magic about 128?

Comment (Catfish-Man) [reply stdlib/public/core/ArrayShared.swift]: Chosen by eyeballing a histogram of string lengths (I was originally doing this change for string, I should probably get a histogram of array counts to confirm it's still reasonable).

**Comment by Catfish-Man** [comment]
@swift-ci please smoke benchmark

**Comment by Catfish-Man** [comment]
Well, that's slightly less bad. Still work to do…

**Comment by Catfish-Man** [comment]
Ah ha! Part of the problem is that we don't have fast dealloc support on the builders

**Comment by Catfish-Man** [comment]
@swift-ci please smoke benchmark

**Comment by Catfish-Man** [comment]
Well, that mostly fixed code size at least :/

**Comment by shahmishal** [comment]
Please update the base branch to `main` by Oct 5th otherwise the pull request will be closed automatically. 

* How to change the base branch: ([Link](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-base-branch-of-a-pull-request))
* More detail about the branch update: ([Link](https://forums.swift.org/t/updating-branch-names/40412/))

**Pull request closed without merging**